### ISSUE TITLE
Inconsistency in issue label endpoints in rest.js

### ISSUE THREAD

**Issue by kytrinyx** [octokit/discussions#19]
@gr2m I ran into an interesting inconsistency today, and I don't think there's an obvious correct answer.

The TL;DR is that the "add labels to issue" and "remove label from issue" endpoints take similar parameters (`owner`, `repo`, `number`) with a fourth key that differs:
- `addLabels` takes `labels`
- `removeLabel` takes `name`

Individually they read really well, but when lined up next to each other it feels inconsistent.
Would this be better if the used the same word stem (e.g. `name`/`names` or `label`/`labels`)?

**Add labels to an issue**
- [GitHub API docs](https://developer.github.com/v3/issues/labels/#add-labels-to-an-issue)
- [Route definition in `rest.js`](https://github.com/octokit/rest.js/blob/04b78765cb0ba7eabc90a9292b661bbac5ed14b3/lib/routes.json#L2860-L2883)

Parameters:

```js
{
  owner: "hello",
  repo: "world",
  number: 1,
  labels: ["label/a", "label/b"]
}
```

**Remove label from an issue**
- [GitHub API docs](https://developer.github.com/v3/issues/labels/#remove-a-label-from-an-issue)
- [Route definition in `rest.js`](https://github.com/octokit/rest.js/blob/04b78765cb0ba7eabc90a9292b661bbac5ed14b3/lib/routes.json#L2884-L2906)

Parameters:

```js
{
  owner: "fruit",
  repo: "apple",
  number: 1,
  name: "label/a"
}
```

I realize that this is based on the GitHub API docs, which show the DELETE endpoint with `:name`, whereas the POST body takes an input of a bare array (somewhat similar to the inconsistency mentioned in #16).

**Comment by gr2m** [comment]
Yeah I agree it would be nice if we could make these more consistent. I don’t have a strong preference on wether using name/names or label/labels, but I agree that either is better than name/labels.

I think it would probably be simpler to do `label/labels` as it wouldn’t need to require any implementation changes for the REST API, only an update of the docs. Instead of

```
DELETE /repos/:owner/:repo/issues/:number/labels/:name
```

We would have

```
DELETE /repos/:owner/:repo/issues/:number/labels/:label
```

On the client side, we could implement aliases and log deprecation warnings for some time to make the transition easier.

If we want to move forward with it, we should look for more cases that are similar, for example

- [`GET /repos/:owner/:repo/labels/:name`](https://developer.github.com/v3/issues/labels/#get-a-single-label)
- [`DELETE /repos/:owner/:repo/labels/:name`](https://developer.github.com/v3/issues/labels/#delete-a-label)

I planned to do a full audit of the [octokit/rest.js/lib/routes.json](https://github.com/octokit/rest.js/blob/04b78765cb0ba7eabc90a9292b661bbac5ed14b3/lib/routes.json) file anyway if you’d like help to gather more potential cases we could harmonize?

**Comment by kytrinyx** [comment]
Yeah, this feels like something that is worth a full audit rather than piecemeal itty-bitty fixes. I'm happy to help gather more potential cases, though at the moment I've not given it any concerted effort, just randomly came across this one.
